### DISCUSSION TITLE
Office 365 dkim error

### DISCUSSION THREAD

**Discussion by sinnbeck** [postalserver/postal#2395] [Help with using Postal] [answered]
I keep getting this error when sending to office 365

>dkim=fail (signature did not verify) 

Mail works when sending to others, and I have even tested with mxtoolbox and similar which all say my DKIM records are correct.

**Comment by willpower232** [comment, marked as answer]
where are you seeing that error? if is it deep in the headers, can you share as much of the headers as possible?

**Reply by sinnbeck** [reply]
I can see it directly in the header of the mail I get. I am suspecting it is somehow due to the psrp record I was asked to set up? I am not quite sure what its purpose is, but if the mail is sent from it instead of the main domain, I suspect the DKIM will indeed fail

I found this in the header

>Authentication-Results: spf=pass (sender IP is 88.99.190.207) smtp.mailfrom=psrp.survey.aalund.com; dkim=fail (signature did not verify) header.d=survey.aalund.com;dmarc=fail action=oreject header.from=survey.aalund.com;compauth=fail reason=000

The error also mentions dmarc, which I assume would fail as the mail is from psrp.survey.aalund.com but the mail itself isnt using the same domain?

**Reply by sinnbeck** [reply]
Ok this is very very strange! I sent a test mail from the webinterface and it arrives as expected with no DKIM errors.. I dont know where to start debugging this. I hope someone has experienced something similar. I assume the mail somehow is altered somehow?

>smtp.mailfrom=psrp.survey.aalund.com; dkim=pass (signature was verified)

**Reply by sinnbeck** [reply]
I found the issue! Apparently the application was adding an unsubscribe header which was used by the old mail provider. After removing that it all seems to work!

**Reply by willpower232** [reply]
Glad you figured it out!
